### PR TITLE
Activate menu items on every subpage as well

### PR DESCRIPTION
**Pull request by Mister-42**
| Q             | A
| ------------- | ---
| Is bugfix?    | ✔️
| New feature?  | ✔️
| Breaks BC?    | ❌
| Fixed issues  | 

This code works, but is ugly. Anyone got something better?

### CODE REVIEW

File: views/layout/main.php [hunk 1]
```diff
@@ -55,9 +57,9 @@
         ->options(['class' => 'navbar-nav mx-auto'])
         ->items(
             [
-                ['label' => 'Blog', 'url' => $urlGenerator->generate('blog/index')],
+                ['label' => 'Blog', 'url' => $urlGenerator->generate('blog/index'), 'active' => in_array($currentRoute, ['blog/index', 'blog/add', 'blog/archive/index', 'blog/archive/month', 'blog/archive/year', 'blog/edit', 'blog/post', 'blog/tag'])],
```
Comment (samdark) [inline views/layout/main.php:60]:
```suggestion
                ['label' => 'Blog', 'url' => $urlGenerator->generate('blog/index'), 'active' => StringHelper::startsWith($currentRoute, 'blog/')],
```

File: views/layout/main.php [hunk 2]
```diff
@@ -55,9 +57,9 @@
...
                 ['label' => 'Comments Feed', 'url' => $urlGenerator->generate('blog/comment/index')],
-                ['label' => 'Users', 'url' => $urlGenerator->generate('user/index')],
+                ['label' => 'Users', 'url' => $urlGenerator->generate('user/index'), 'active' => in_array($currentRoute, ['user/index', 'user/profile'])],
```
Comment (samdark) [inline views/layout/main.php:62]:
```suggestion
                ['label' => 'Users', 'url' => $urlGenerator->generate('user/index'), 'active' => StringHelper::startsWith($currentRoute, 'user/')],
```

File: views/layout/main.php (hunk 1 above)
Comment (samdark) [reply views/layout/main.php:60]: But don't forget to `use` the helper.

**Review by samdark — CHANGES_REQUESTED** [review]
Using string match would allow you not to modify condition when you add another sub-route.

File: views/layout/main.php (hunk 1 above)
Comment (Mister-42) [reply views/layout/main.php:60]: Actually, that would also pull in `blog/comment/index`, which I excluded on purpose (it's the next menu entry)

Comment (samdark) [reply views/layout/main.php:61]: Well, for that `!== 'blog/comment/index'` condition could be added. Still better than keeping the list up to date.

Comment (Mister-42) [reply views/layout/main.php:61]: Fair enough. I was not aware that part exists 👍🏻

Comment (Mister-42) [reply views/layout/main.php:61]: Fair enough

**Review by samdark — APPROVED** [review]

**Comment by samdark** [comment]
👍

**Pull request merged**